iothub/device/src/IotHubBaseClient.cs

```diff
@@ -79,6 +79,11 @@ public abstract class IotHubBaseClient : IDisposable
         /// <summary>
         /// The callback to be executed each time connection status change notification is received.
         /// </summary>
+        /// <remarks>
+        /// This user-supplied callback is "fire-and-forget" and the SDK doesn't wait on it.
```

Comment:
Do we have reference or document to point out what are the exceptions the user can expect?
What about updating the example with 'try/catch' to explicitly remind the user about handling exceptions?

Reply:
We call the user's callback, therefore we can't throw exceptions that would impact them here. However, if they throw an exception, we need to handle it so as to not crash the process. We're just fixing and documenting that behavior.

Reply:
This specific callback doesn't return anything (`Action<ConnectionStatusInfo>`), so instead of calling this fire-and-forget (which would mean that it returns a task but we don't await on it) why don't we call out that only if you put async code within this callback, you'll need to handle exceptions that could originate in there?